fdbclient/admin_client.go

```diff
@@ -318,7 +308,30 @@ func (client *cliAdminClient) getStatus(useClientLibrary bool) (*fdbv1beta2.Foun
 	if err != nil {
 		return nil, err
 	}
-	client.log.V(1).Info("Fetched status JSON", "status", status)
+
+	client.log.V(1).Info("Fetched status JSON with fdbcli", "version", client.Cluster.GetRunningVersion(), "status", status)
+
+	return status, nil
+}
+
+// getStatus uses fdbcli to connect to the FDB cluster, if the cluster is upgraded and the initial version returns no processes
```

Comment:
You don't want to handle the partial upgrade scenario (where a subset of processes are still/stuck at the older version for whatever reason) here?

It appears you are trying to implement (a version of) the multiversion client logic here, probably because of a limitation of that logic? Maybe you can explain that limitation and how you are addressing that here (so we can follow what you are trying to do)? Thanks!

Reply:
> You don't want to handle the partial upgrade scenario (where a subset of processes are still/stuck at the older version for whatever reason) here?

What scenario do you have in mind? As far as I understand the logic in FDB the majority of coordinators will define the version of the cluster e.g. a cluster can only be at one version (and have a few other processes running at a different version but not reporting to the cluster). We observed a few cases where the kill was not restarting one coordinator (for different reasons) and that resulted in the multi version client behaving badly.

> It appears you are trying to implement (a version of) the multiversion client logic here, probably because of a limitation of that logic? Maybe you can explain that limitation and how you are addressing that here (so we can follow what you are trying to do)? Thanks!

We are implementing it because the multi version logic is currently broken (I'm not sure if that is limited to 6.3 and is fixed with 7.1), there are some comments about this in the FDB Slack. Basically by using the fdbcli binary without multi version configuration we force the operator to use a specific version to connect to the DB.